### ISSUE TITLE
CC 3.0 not supported since 1.0.0-beta4

### ISSUE THREAD

**Issue by MichalZelechowski** [eclipse/deeplearning4j#8018]
#### Issue Description

When ND4j is initialized with CUDA profile 3.0 `cudaGetSymbolAdress(...)` fails.

There's no such error with 1.0.0-beta3. Seems that since 1.0.0-beta4 libnd4j does not support CC profiles <= 3.0. When trying to compile libnd4j with `-cc 30` the error is reported:
`
libnd4j/include/templatemath.h(789): error: no instance of overloaded function "atomicMin" matches the argument list
             argument types are: (unsigned long long *, unsigned long long)
`
According to `cuda-c-programming-guide`:
`
The 64-bit version of atomicMin() is only supported by devices of compute capability 3.5 and higher. 
`
Lack of support of ccs<=3.0 excludes wide range of GPUs from GTX 7xx line. It would be great to have this support back.

#### Version Information

* Deeplearning4j version: 1.0.0-beta4
* Platform information: win 10, linux
* CUDA version: 10.0.130, 10.1.168
* NVIDIA driver version: 418.67, 431.36

#### Additional Information

Example stack trace from running `MNISTClassifier`:
`
o.d.e.c.m.MnistClassifier - Data load... o.d.e.c.m.MnistClassifier - Data vectorization... o.d.i.r.BaseImageRecordReader - ImageRecordReader: 10 label classes inferred using label generator ParentPathLabelGenerator o.d.i.r.BaseImageRecordReader - ImageRecordReader: 10 label classes inferred using label generator ParentPathLabelGenerator o.d.e.c.m.MnistClassifier - Network configuration and training... o.n.l.f.Nd4jBackend - Loaded [JCublasBackend] backend o.n.n.NativeOpsHolder - Number of threads used for NativeOps: 32 o.n.n.Nd4jBlas - Number of threads used for BLAS: 0 o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CUDA]; OS: [Windows 10] o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [3,5GB]; o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [CUBLAS] o.n.l.j.o.e.CudaExecutioner - Device Name: [GeForce GTX 770]; CC: [3.0]; Total/free memory: [2147483648] o.d.n.m.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE] Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: cudaGetSymbolAddress(...) failed at org.nd4j.jita.constant.ProtectedCudaConstantHandler.ensureMaps(ProtectedCudaConstantHandler.java:308) at org.nd4j.jita.constant.ProtectedCudaConstantHandler.moveToConstantSpace(ProtectedCudaConstantHandler.java:124) at org.nd4j.jita.constant.CudaConstantHandler.moveToConstantSpace(CudaConstantHandler.java:42) at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:96) at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:73) at org.nd4j.linalg.jcublas.CachedShapeInfoProvider.createShapeInformation(CachedShapeInfoProvider.java:42) at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:216) at org.nd4j.linalg.jcublas.JCublasNDArray.<init>(JCublasNDArray.java:382) at org.nd4j.linalg.jcublas.JCublasNDArrayFactory.create(JCublasNDArrayFactory.java:1655) at org.nd4j.linalg.jcublas.JCublasNDArrayFactory.create(JCublasNDArrayFactory.java:1650) at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:4651) at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:4129) at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.init(MultiLayerNetwork.java:681) at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.init(MultiLayerNetwork.java:600) at org.deeplearning4j.examples.convolution.mnist.MnistClassifier.main(MnistClassifier.java:160) Caused by: java.lang.RuntimeException: cudaGetSymbolAddress(...) failed at org.nd4j.nativeblas.Nd4jCuda$NativeOps.getConstantSpace(Native Method) at org.nd4j.jita.constant.ProtectedCudaConstantHandler.ensureMaps(ProtectedCudaConstantHandler.java:301) ... 14 more
`

**Comment by raver119** [comment]
TLDR: we need to implement few other atomic methods not supported at lower CCs.

**Comment by raver119** [comment]
cc @shugeo :)

**Comment by raver119** [comment]
Fixed.

**Comment by GPSforLEGENDS** [comment]
This is still not fixed. Atleast not for my gtx 770:
`o.d.e.f.m.MLPMnistTwoLayerExample - Build model....
o.n.l.f.Nd4jBackend - Loaded [JCublasBackend] backend
ND4J CUDA build version: 10.0.130
CUDA device 0: [GeForce GTX 770]; cc: [3.0]; Total memory: [2147483648];
o.n.n.NativeOpsHolder - Number of threads used for OpenMP: 32
o.n.n.Nd4jBlas - Number of threads used for OpenMP BLAS: 0
o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CUDA]; OS: [Windows 10]
o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [3,5GB];
o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [CUBLAS]
o.n.l.j.o.e.CudaExecutioner - Device Name: [GeForce GTX 770]; CC: [3.0]; Total/free memory: [2147483648]
o.d.n.m.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE]
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.CommandLineWrapper.main(CommandLineWrapper.java:66)
Caused by: java.lang.RuntimeException: cudaGetSymbolAddress(...) failed; Error code: [13]
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.createShapeInfo(CudaExecutioner.java:2557)
	at org.nd4j.linalg.api.shape.Shape.createShapeInformation(Shape.java:3282)
	at org.nd4j.linalg.api.ndarray.BaseShapeInfoProvider.createShapeInformation(BaseShapeInfoProvider.java:76)
	at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:96)
	at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:77)
	at org.nd4j.linalg.jcublas.CachedShapeInfoProvider.createShapeInformation(CachedShapeInfoProvider.java:44)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:211)
	at org.nd4j.linalg.jcublas.JCublasNDArray.<init>(JCublasNDArray.java:383)
	at org.nd4j.linalg.jcublas.JCublasNDArrayFactory.create(JCublasNDArrayFactory.java:1543)
	at org.nd4j.linalg.jcublas.JCublasNDArrayFactory.create(JCublasNDArrayFactory.java:1538)
	at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:4298)
	at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:3986)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.init(MultiLayerNetwork.java:688)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.init(MultiLayerNetwork.java:604)
	at org.deeplearning4j.examples.feedforward.mnist.MLPMnistTwoLayerExample.main(MLPMnistTwoLayerExample.java:100)
	... 5 more`

**Comment by MichalZelechowski** [comment]
Same with my configuration:

`o.n.l.f.Nd4jBackend - Loaded [JCublasBackend] backend
ND4J CUDA build version: 10.1.243
CUDA device 0: [GeForce GTX 760]; cc: [3.0]; Total memory: [2093809664];
o.n.n.NativeOpsHolder - Number of threads used for OpenMP: 32
o.n.n.Nd4jBlas - Number of threads used for OpenMP BLAS: 0
o.n.l.a.o.e.DefaultOpExecutioner - Backend used: [CUDA]; OS: [Linux]
o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [3,5GB];
o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [CUBLAS]
o.n.l.j.o.e.CudaExecutioner - Device Name: [GeForce GTX 760]; CC: [3.0]; Total/free memory: [2093809664]
Exception in thread "main" java.lang.RuntimeException: cudaGetSymbolAddress(...) failed; Error code: [13]
	at org.nd4j.linalg.jcublas.ops.executioner.CudaExecutioner.createShapeInfo(CudaExecutioner.java:2557)
	at org.nd4j.linalg.api.shape.Shape.createShapeInformation(Shape.java:3282)
	at org.nd4j.linalg.api.ndarray.BaseShapeInfoProvider.createShapeInformation(BaseShapeInfoProvider.java:76)
	at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:96)
	at org.nd4j.jita.constant.ProtectedCudaShapeInfoProvider.createShapeInformation(ProtectedCudaShapeInfoProvider.java:77)
	at org.nd4j.linalg.jcublas.CachedShapeInfoProvider.createShapeInformation(CachedShapeInfoProvider.java:44)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:205)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:200)
	at org.nd4j.linalg.jcublas.JCublasNDArray.<init>(JCublasNDArray.java:375)
	at org.nd4j.linalg.jcublas.JCublasNDArrayFactory.create(JCublasNDArrayFactory.java:1518)
	at org.nd4j.linalg.factory.BaseNDArrayFactory.create(BaseNDArrayFactory.java:1160)
	at org.nd4j.linalg.factory.BaseNDArrayFactory.create(BaseNDArrayFactory.java:1224)
	at org.nd4j.linalg.factory.BaseNDArrayFactory.zeros(BaseNDArrayFactory.java:993)
	at org.nd4j.linalg.factory.BaseNDArrayFactory.zeros(BaseNDArrayFactory.java:735)
	at org.nd4j.linalg.factory.Nd4j.zeros(Nd4j.java:4455)
	at org.nd4j.examples.Nd4jEx1_INDArrayBasics.main(Nd4jEx1_INDArrayBasics.java:47)`

Though was indeed able to build branch **release/1.0.0-beta5** locally with **CC profile 3.0**.
